src/apel_plugin.py

```diff
@@ -197,40 +197,44 @@ async def create_summary_db(config, records):
         logging.critical(e)
         raise e
 
-    sites_to_report = config["site"].get("sites_to_report")
-
     try:
         site_name_mapping = json.loads(config["site"].get("site_name_mapping"))
     except TypeError:
         site_name_mapping = None
 
+    sites_to_report = config["site"].get("sites_to_report")
     vo_mapping = json.loads(config["uservo"].get("vo_mapping"))
     submit_host = config["site"].get("submit_host")
     infrastructure = config["site"].get("infrastructure_type")
     benchmark_name = config["auditor"].get("benchmark_name")
     cores_name = config["auditor"].get("cores_name")
     cpu_time_name = config["auditor"].get("cpu_time_name")
     nnodes_name = config["auditor"].get("nnodes_name")
+    meta_key_site = config["auditor"].get("meta_key_site")
+    meta_key_user = config["auditor"].get("meta_key_user")
 
     for r in records:
+        site_id = r.meta.get(meta_key_site)[0]
+        user_id = r.meta.get(meta_key_user)[0]
+
         if sites_to_report != "all":
-            if r.site_id not in json.loads(sites_to_report):
+            if site_id not in json.loads(sites_to_report):
```

Comment:
`json.loads(...)` is executed for every record, which seems wasteful. Could you move this before the loop?

Reply:
Done with [bd25b67](https://github.com/ALU-Schumacher/AUDITOR-APEL-plugin/pull/37/commits/bd25b67293e6498bd0338b80e891462de92ed1b2)
I also removed the option "all", which simplifies the logic.